### PR TITLE
[ENHANCEMENT] Streamline package-manager CLI options

### PR DESCRIPTION
**Pull request by bertdeblock**
At the moment, the `ember install` command supports a `package-manager` option, with `pnpm` and `yarn` as value aliases, whereas the `ember addon`, `ember init` and `ember new` commands have separate `pnpm` and `yarn` options, and don't support the `package-manager` option.

This PR adds support for the `package-manager` option to all Ember CLI commands with `npm`, `pnpm` and `yarn` as value aliases.

This PR also fixes a few things:

- Use pnpm when generating a new Ember CLI project inside a pnpm workspace
- Skip the package-manager question during the interactive-new flow, when using the `--pnpm` option
- Use the correct package-manager install command in errors (`npm install` vs `pnpm install` vs `yarn install`)

### CODE REVIEW

File: blueprints/addon/index.js [hunk 1]
```diff
@@ -161,8 +161,8 @@ module.exports = {
         indent +
         [
           options.welcome && '"--welcome"',
-          options.yarn && '"--yarn"',
-          options.pnpm && '"--pnpm"',
+          options.packageManager === 'yarn' && '"--yarn"',
```
Comment (bertdeblock) [inline blueprints/addon/index.js:164]: Could also do:

```js
options.packageManager && `"--package-manager=${options.packageManager}"`,
```

But seems more disruptive. Let me know what you think.

File: lib/commands/init.js [hunk 2]
```diff
@@ -80,12 +82,16 @@ module.exports = Command.extend({
       commandOptions.lang = getLangArg(commandOptions.lang, this.ui);
     }
 
-    let blueprintOpts = clone(commandOptions);
-
-    if (blueprintOpts.yarn === undefined && project.isEmberCLIProject()) {
-      blueprintOpts.yarn = isYarnProject(project.root);
+    if (commandOptions.packageManager === undefined && project.isEmberCLIProject()) {
```
Comment (bertdeblock) [inline lib/commands/init.js:85]: pnpm check was missing here.

File: lib/commands/install.js [hunk 3]
```diff
@@ -20,18 +21,19 @@ module.exports = Command.extend({
...
   ],
 
   anonymousOptions: ['<addon-name>'],
 
-  run(commandOptions, addonNames) {
+  async run(commandOptions, addonNames) {
     if (!addonNames.length) {
-      let msg = 'The `install` command must take an argument with the name';
-      msg += ' of an ember-cli addon. For installing all npm ';
-      msg += 'dependencies you can run `npm install`.';
```
Comment (bertdeblock) [inline lib/commands/install.js:33]: `npm install` was still hardcoded here. Should now use the correct package manager.

File: lib/models/command.js [hunk 4]
```diff
@@ -256,60 +256,60 @@ let Command = CoreObject.extend({
...
   */
-  validateAndRun(args) {
-    return new Promise((resolve, reject) => {
-      let commandOptions = this.parseArgs(args);
-      // if the help option was passed, resolve with 'callHelp' to call help command
-      if (commandOptions && (commandOptions.options.help || commandOptions.options.h)) {
-        logger.info(`${this.name} called with help option`);
-        return resolve('callHelp');
-      }
+  async validateAndRun(args) {
```
Comment (bertdeblock) [inline lib/models/command.js:259]: Sorry for the whitespace changes.
Had to make this function async to use the `determineInstallCommand` helper.

File: lib/tasks/interactive-new.js [hunk 5]
```diff
@@ -81,7 +81,7 @@ class InteractiveNewTask extends Task {
         name: 'packageManager',
         type: 'list',
         message: 'Pick the package manager to use when installing dependencies:',
-        when: !newCommandOptions.yarn,
+        when: !newCommandOptions.packageManager,
```
Comment (bertdeblock) [inline lib/tasks/interactive-new.js:84]: Take `--npm` and `--pnpm` into account as well for skipping this question.

File: lib/tasks/npm-task.js [hunk 6]
```diff
@@ -50,8 +47,8 @@ class NpmTask extends Task {
     return isYarnProject(this.project.root);
   }
 
-  async hasPNPMLock() {
-    return Boolean(await findUp('pnpm-lock.yaml', { cwd: this.project.root }));
+  hasPNPMLock() {
+    return isPnpmProject(this.project.root);
```
Comment (bertdeblock) [inline lib/tasks/npm-task.js:51]: Just to be consistent in how we determine if a project is a pnpm project.

File: lib/commands/addon.js [hunk 7]
```diff
@@ -13,8 +13,11 @@ module.exports = NewCommand.extend({
     { name: 'blueprint', type: String, default: 'addon', aliases: ['b'] },
     { name: 'skip-npm', type: Boolean, default: false, aliases: ['sn'] },
     { name: 'skip-git', type: Boolean, default: false, aliases: ['sg'] },
-    { name: 'yarn', type: Boolean }, // no default means use yarn if the blueprint has a yarn.lock
-    { name: 'pnpm', type: Boolean, default: false },
+    {
+      name: 'package-manager',
+      type: ['npm', 'pnpm', 'yarn'],
+      aliases: [{ npm: 'npm' }, { pnpm: 'pnpm' }, { yarn: 'yarn' }],
```
Comment (NullVoxPopuli) [inline lib/commands/addon.js:19]: ooo fancy

File: lib/tasks/interactive-new.js (hunk 5 above)
Comment (NullVoxPopuli) [reply lib/tasks/interactive-new.js:84]: yaaas

**Review by NullVoxPopuli — APPROVED** [review]

**Review by kategengler — APPROVED** [review]

**Pull request merged**